From cde86a0561724b29248b9c6f887d4c40a8a32277 Mon Sep 17 00:00:00 2001
From: Radha Mohan Chintakuntla <radhac@marvell.com>
Date: Thu, 15 Oct 2020 16:47:24 -0700
Subject: [PATCH 12/19] soc: octeontx2-sdp: Fix ring info structure that is
 exchanged with host

commit dde73a620a0d4c092c59e54c12798eaa8f826f82 from
git@git.assembla.com:cavium/WindRiver.linux.git

The host side drivers have been updated of the ring info structure but
EP side wasn't. This was causing issues in negotiating ring and VF
information. This patch fixes the issue by sync'ing the structure.

Signed-off-by: Radha Mohan Chintakuntla <radhac@marvell.com>
Change-Id: I2ec7df581483a4686678f4448ef5c10fb4f4e3ad
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/38025
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Abhijit Ayarekar <aayarekar@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/49037
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/soc/marvell/octeontx2-sdp/sdp.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/soc/marvell/octeontx2-sdp/sdp.c b/drivers/soc/marvell/octeontx2-sdp/sdp.c
index 302b9709211b..b2650d1e371c 100644
--- a/drivers/soc/marvell/octeontx2-sdp/sdp.c
+++ b/drivers/soc/marvell/octeontx2-sdp/sdp.c
@@ -39,10 +39,10 @@ union ring {
 	u64 u;
 	struct {
 		u64 dir:2;
-		u64 rpvf:4;
-		u64 rppf:6;
+		u64 rpvf:8;
+		u64 rppf:8;
 		u64 numvf:8;
-		u64 rsvd:16;
+		u64 rsvd:10;
 		u64 raz:28;
 	} s;
 };
-- 
2.31.1

